samcli/commands/build/utils.py

```diff
@@ -1,13 +1,65 @@
 """
 Utilities for sam build command
 """
+import pathlib
 
 import click
 
-from samcli.lib.build.workflow_config import CONFIG
+from samcli.lib.build.workflow_config import CONFIG, get_workflow_config
+from samcli.lib.providers.provider import ResourcesToBuildCollector
 
 
-def prompt_user_to_enable_mount_with_write(config: CONFIG, source_dir: str) -> bool:
+def prompt_user_to_enable_mount_with_write_if_needed(
+    resources_to_build: ResourcesToBuildCollector,
+    base_dir: str,
+) -> bool:
+    """
+    First check if mounting with write permissions is needed for building inside container or not. If it is needed, then
+    prompt user to choose if enables mounting with write permissions or not.
+
+    Parameters
+    ----------
+    resources_to_build:
+        Resource to build inside container
+
+    base_dir : str
+        Path to the base directory
+
+    Returns
+    -------
+    bool
+        True, if user enabled mounting with write permissions.
+    """
+
+    functions = resources_to_build.functions
+    layers = resources_to_build.layers
+    runtime = None
+    specified_workflow = None
+    code_uri = None
+
+    if len(functions) > 0:
+        function = functions[0]
+        runtime = function.runtime
+        code_uri = function.codeuri
+        # get specified_workflow if metadata exists
+        metadata = function.metadata
+        specified_workflow = metadata.get("BuildMethod", None) if metadata else None
+    elif len(layers) > 0:
+        layer = layers[0]
+        specified_workflow = layer.build_method
+        code_uri = layer.codeuri
```

Comment:
Why we are checking just one function and layer? Shouldn't this loop through all of them to see if any of them satisfy the need for the prompt?